### PR TITLE
Trying again to fix my formatting.

### PR DESCRIPTION
**Pull request by madams51703**
Ok, here is my latest attempt to fix my formatting.  Starting to wrap my head around the proper format, but am sure there is a lot for me to learn.  Here goes nothing! :+)

### CODE REVIEW

**Comment by KvanTTT** [comment]
There are some formatting issues, but ok, will fix it later.

**Comment by madams51703** [comment]
Please advise on any additional formatting rules so I can try to learn them
so I do not reintroduce them and continue to to try to fix the ones I have
introduced.  Am trying to learn from the rules you layed out.  I am slowly
trying to wrap my head around it.

Thanks!!

File: tsql/TSqlParser.g4 [hunk 1]
```diff
@@ -205,54 +205,108 @@ another_statement
...
-       (PASSWORD EQUAL  application_role_password=STRING COMMA?)?
-       (DEFAULT_SCHEMA  EQUAL app_role_default_schema=id COMMA?)?
+    : alter_application_role_with
+       (alter_application_role_with_name new_application_role_name=id COMMA?)
+       (alter_application_role_with_password  application_role_password=STRING COMMA?)?
+       (alter_application_role_with_default app_role_default_schema=id COMMA?)?
+    ;
+
+alter_application_role_with
+    : WITH
```
Comment (KvanTTT) [inline tsql/TSqlParser.g4]: Please do not use parse rules for single tokens.

Remove
```ANTLR
alter_application_role_with
    : WITH
    ;
```

And paste `WITH` keyword (ant other) to `alter_application_role_with_clause`:
```ANTLR
 alter_application_role_with_clause
     : WITH
       (NAME EQUAL new_application_role_name=id COMMA?)
       (PASSWORD EQUAL  application_role_password=STRING COMMA?)?
       (DEFAULT_SCHEMA EQUAL app_role_default_schema=id COMMA?)?
    ;
```

This improves readability and simplifies grammar.
This note related to other your changes too.

File: tsql/TSqlParser.g4 [hunk 2]
```diff
@@ -205,54 +205,108 @@ another_statement
...
 alter_assembly_add_clause
-    : ADD FILE FROM client_file_clause 
+    : alter_asssembly_add_clause_start alter_assembly_client_file_clause 
     ;
 
+alter_asssembly_add_clause_start
+    : ADD FILE FROM
+    ;
+   
+
```
Comment (KvanTTT) [inline tsql/TSqlParser.g4]: Excess line break.

File: tsql/TSqlParser.g4 [hunk 3]
```diff
@@ -261,49 +315,69 @@ client_assembly_specifier
...
 local_file
-    : DISK_DRIVE file_path
+    : local_drive file_path
     ;
 
+local_drive
+    :
+    DISK_DRIVE
+    ;
```
Comment (KvanTTT) [inline tsql/TSqlParser.g4]: Incorrect formatting. Use the following instead (the same for other such rules):
```ANTLR
    : DISK_DRIVE
    ;
```

File: tsql/TSqlParser.g4 [hunk 4]
```diff
@@ -781,134 +873,204 @@ cursor_option:
...
+   ;
 
 ipv4_address:
    IPV4_ADDR  
    ;
 
 
 ipv6_address:
-        IPV6_ADDR
-	;
```
Comment (KvanTTT) [inline tsql/TSqlParser.g4]: Do not use tabs. Use whitespaces everywhere.

**Comment by madams51703** [comment]
Thank you for the feedback.  I think my next pull request that I am going
to put up shortly is better formatted and more simple.



On Jul 28, 2017 15:40, "Ivan Kochurkin" <notifications@github.com> wrote:

*@KvanTTT* commented on this pull request.

I added some review notes:

   - Rules simplification.
   - Rules formatting.
   - Excess line breaks.
   - Use whitespaces instead of tabs.

------------------------------

In tsql/TSqlParser.g4
<https://github.com/antlr/grammars-v4/pull/816#discussion_r130166528>:

>      ;

 alter_application_role_with_clause
-    :WITH
-       (NAME EQUAL new_application_role_name=id COMMA?)
-       (PASSWORD EQUAL  application_role_password=STRING COMMA?)?
-       (DEFAULT_SCHEMA  EQUAL app_role_default_schema=id COMMA?)?
+    : alter_application_role_with
+       (alter_application_role_with_name new_application_role_name=id COMMA?)
+       (alter_application_role_with_password
application_role_password=STRING COMMA?)?
+       (alter_application_role_with_default app_role_default_schema=id COMMA?)?
+    ;
+
+alter_application_role_with
+    : WITH

Please do not use parse rules for single tokens.

Remove

alter_application_role_with
    : WITH
    ;

And paste WITH keyword (ant other) to alter_application_role_with_clause:

 alter_application_role_with_clause
     : WITH
       (NAME EQUAL new_application_role_name=id COMMA?)
       (PASSWORD EQUAL  application_role_password=STRING COMMA?)?
       (DEFAULT_SCHEMA EQUAL app_role_default_schema=id COMMA?)?
    ;

This improves readability and simplifies grammar.
This note related to other your changes too.
------------------------------

In tsql/TSqlParser.g4
<https://github.com/antlr/grammars-v4/pull/816#discussion_r130166928>:

>      ;

+alter_asssembly_add_clause_start
+    : ADD FILE FROM
+    ;
+
+

Excess line break.
------------------------------

In tsql/TSqlParser.g4
<https://github.com/antlr/grammars-v4/pull/816#discussion_r130167110>:

>      ;

+local_drive
+    :
+    DISK_DRIVE
+    ;

Incorrect formatting. Use the following instead (the same for other such
rules):

    : DISK_DRIVE
    ;

------------------------------

In tsql/TSqlParser.g4
<https://github.com/antlr/grammars-v4/pull/816#discussion_r130167444>:

>
 ipv4_address:
    IPV4_ADDR
    ;


 ipv6_address:
-        IPV6_ADDR
-	;

Do not use tabs. Use whitespaces everywhere.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/antlr/grammars-v4/pull/816#pullrequestreview-53016948>,
or mute the thread

**Pull request merged**